x-pack/plugins/timelines/server/search_strategy/timeline/factory/events/all/query.events_all.dsl.ts

```diff
@@ -58,7 +58,7 @@ export const buildTimelineEventsAllQuery = ({
       return {
         [field]: {
           order: item.direction,
-          unmapped_type: item.type,
+          unmapped_type: 'keyword',
```

Comment:
To desk test this change, I added a `console.log` to the following code in `x-pack/plugins/timelines/server/search_strategy/timeline/factory/events/all/query.events_all.dsl.ts` (in `main`):

```ts
  const getSortField = (sortFields: TimelineRequestSortField[]) =>
    sortFields.map((item) => {
      const field: string = item.field === 'timestamp' ? '@timestamp' : item.field;
      console.log('---> item', item);
      return {
        [field]: {
          order: item.direction,
          unmapped_type: item.type,
        },
      };
    });
```

to visualize the `item`, and then clicked to add a (secondary) sort on the `event.action` column in a timeline.

When I clicked that field, the `console.log` emitted (in `main`) the following output:

```
---> item { field: '@timestamp', direction: 'asc', type: 'date' }
---> item { field: 'event.action', direction: 'desc', type: 'string' }
```

and the `sort` part of the query DSL (when inspected) looks like this:

```json
  "sort": [
    {
      "@timestamp": {
        "order": "asc",
        "unmapped_type": "date"
      }
    },
    {
      "event.action": {
        "order": "desc",
        "unmapped_type": "string"
      }
    }
  ],
```

The `event.action` field is defined as a `keyword` (not `string`) in ECS [here](https://github.com/elastic/ecs/blob/051e91ba9ad78934d8d963bca7f7b09ef647b96c/schemas/event.yml#L351), so it's reasonable to expect that, for the `event.action` field, the `sort` entry would instead use ` "unmapped_type": "keyword"` instead of  "unmapped_type": "string".

With the proposed change, the `event.action` `sort` entry now has the expected `"unmapped_type": "keyword"`, but a side-effect of this change is the `@timestamp` field will also have an `"unmapped_type": "keyword"` entry (instead of `"unmapped_type": "date"` in `main`), per the inspected query output below:

```
  "sort": [
    {
      "@timestamp": {
        "order": "asc",
        "unmapped_type": "keyword"
      }
    },
    {
      "event.action": {
        "order": "desc",
        "unmapped_type": "keyword"
      }
    }
  ],
```

It fees like the spirit of the original code, using `unmapped_type: item.type` is correct, but the `item.type` metadata for some fields, like `event.action` in the example below, is not granular enough, because it's a `string` instead of a `keyword`:

```
---> item { field: '@timestamp', direction: 'asc', type: 'date' }
---> item { field: 'event.action', direction: 'desc', type: 'string' }
```

Reply:
Even though `event.action` is defined by ECS as a `keyword`, when I view `auditbeat` data in Discover, and hover over the `event.action` field, it also displays a type of `String field`, per the screenshot below:

![discover-auditbeat-event-action](https://user-images.githubusercontent.com/4459398/162280415-17b5d7fa-4aa5-4eb2-9a24-c8a103f23a9d.png)

Reply:
In the following screenshot obtained by `console.log`ing the browser fields metadata for `event.action`:

![browserFIeldsEventAction](https://user-images.githubusercontent.com/4459398/162350673-3ac12c46-0e08-4c5c-8647-51ad91df11ab.png)

- The `type` property in the metadata above is `string`
- The first / only entry in the `esTypes` array is `keyword`, which is consistent with the ECS definition of `event.action`
- The `type` property is currently displayed in the column tooltip
- Given the output below includes the `direction: 'asc'` property, it appears that the column metadata may also be the `item` used in the context of the `getSortField` function to request `unmapped_type: item.type`:

```
---> item { field: '@timestamp', direction: 'asc', type: 'date' }
---> item { field: 'event.action', direction: 'desc', type: 'string' }
```

Given the above:

- Consider updating the column metadata to use the values from the `esTypes` array, instead of `type`
- Given that `esTypes` is an array, this change would require updating both the column tooltip, and the request currently using `unmapped_type: item.type` to handle the case when the array contains more than one type